files/image_config/platform/rc.local

```diff
@@ -11,7 +11,10 @@
 #
 # By default this script does nothing.
 
-SONIC_VERSION=$(sonic-cfggen -y /etc/sonic/sonic_version.yml -v build_version)
+# read SONiC immutable variables
+[ -f /etc/sonic/sonic-environment ] && . /etc/sonic/sonic-environment
+
+SONIC_VERSION=${SONIC_VERSION:-`sonic-cfggen -y /etc/sonic/sonic_version.yml -v build_version`}
```

Comment:
Why replace the `$(...)` syntax with backticks?

Reply:
This [parameter substitution syntax](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02) and I could embed $() in it.

Reply:
>I could embed $() in it.

I assume you mean "couldn't"?

Reply:
updates comment. I could not.